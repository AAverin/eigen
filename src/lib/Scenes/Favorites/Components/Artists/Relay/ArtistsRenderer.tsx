import React from "react"
import { graphql, QueryRenderer, QueryRendererProps } from "react-relay"

import createEnvironment from "lib/relay/createEnvironment"
const environment = createEnvironment()

export default ({ render }) => {
  return (
    <QueryRenderer
      environment={environment}
      query={graphql`
        query ArtistsRendererQuery {
          me {
            ...Artists_me
          }
        }
      `}
      variables={null}
      render={render}
    />
  )
}
