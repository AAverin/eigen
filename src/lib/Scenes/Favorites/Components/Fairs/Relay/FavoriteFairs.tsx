import React from "react"
import { graphql, QueryRenderer } from "react-relay"

import { FavoriteFairsQuery } from "__generated__/FavoriteFairsQuery.graphql"
import createEnvironment from "lib/relay/createEnvironment"
const environment = createEnvironment()

export default ({ render }) => {
  return (
    <QueryRenderer<FavoriteFairsQuery>
      environment={environment}
      query={graphql`
        query FavoriteFairsQuery {
          me {
            ...Fairs_me
          }
        }
      `}
      variables={{
        count: 10,
      }}
      render={render}
      cacheConfig={{ force: true }}
    />
  )
}
