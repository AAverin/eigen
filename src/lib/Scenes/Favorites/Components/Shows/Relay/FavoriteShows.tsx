import React from "react"
import { graphql, QueryRenderer } from "react-relay"

import { FavoriteShowsQuery } from "__generated__/FavoriteShowsQuery.graphql"
import createEnvironment from "lib/relay/createEnvironment"
const environment = createEnvironment()

export default ({ render }) => {
  return (
    <QueryRenderer<FavoriteShowsQuery>
      environment={environment}
      query={graphql`
        query FavoriteShowsQuery {
          me {
            ...Shows_me
          }
        }
      `}
      variables={{
        count: 10,
      }}
      render={render}
      cacheConfig={{ force: true }}
    />
  )
}
