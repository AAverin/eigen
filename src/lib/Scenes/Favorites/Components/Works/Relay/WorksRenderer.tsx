import React from "react"
import { graphql, QueryRenderer, QueryRendererProps } from "react-relay"

import createEnvironment from "lib/relay/createEnvironment"
const environment = createEnvironment()

export default ({ render }) => {
  return (
    <QueryRenderer
      environment={environment}
      query={graphql.experimental`
        query WorksRendererQuery($count: Int!, $cursor: String) {
          me {
            ...Works_me @arguments(count: $count, cursor: $cursor)
          }
        }
      `}
      variables={{
        count: 10,
      }}
      render={render}
    />
  )
}
