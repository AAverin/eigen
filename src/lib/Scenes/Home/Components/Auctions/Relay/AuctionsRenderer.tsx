import * as React from "react"
import { graphql, QueryRenderer, QueryRendererProps } from "react-relay"

import createEnvironment from "../../../../../relay/createEnvironment"
const environment = createEnvironment()

const AuctionsRenderer: React.SFC<any> = ({ render }) => {
  return (
    <QueryRenderer
      environment={environment}
      query={graphql`
        query AuctionsRendererQuery {
          auctions: sales(live: true, is_auction: true) {
            ...Auctions_auctions
          }
        }
      `}
      variables={{}}
      render={render}
    />
  )
}

export default AuctionsRenderer
