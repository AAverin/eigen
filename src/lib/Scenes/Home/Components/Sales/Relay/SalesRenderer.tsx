import { defaultEnvironment } from "lib/relay/createEnvironment"
import React from "react"
import { graphql, QueryRenderer } from "react-relay"
import { PAGE_SIZE } from "../Components/LotsByFollowedArtists"

export function SalesRenderer({ render }) {
  return (
    <QueryRenderer
      environment={defaultEnvironment}
      query={graphql.experimental`
        query SalesRendererQuery {
          sales: sales(live: true, is_auction: true) {
            ...Sales_sales
          }
          viewer {
            ...Sales_viewer
          }
        }
      `}
      variables={{
        count: PAGE_SIZE,
      }}
      render={render}
    />
  )
}
