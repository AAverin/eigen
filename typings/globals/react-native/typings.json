{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
