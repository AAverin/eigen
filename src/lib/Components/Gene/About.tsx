import React from "react"
import { createFragmentContainer, graphql } from "react-relay"

import { StyleSheet, ViewStyle } from "react-native"

import Biography from "./Biography"

import RelatedArtists from "../RelatedArtists/RelatedArtists"
import Separator from "../Separator"

import { About_gene } from "__generated__/About_gene.graphql"
import { StickyTabPageScrollView } from "../StickyTabPage/StickyTabPageScrollView"

interface Props {
  gene: About_gene
}

const About: React.FC<Props> = ({ gene }) => {
  const relatedArtists = () => {
    return (gene.trending_artists || []).length ? <RelatedArtists artists={gene.trending_artists as any} /> : null
  }

  const biography = () => {
    return (
      <>
        <Biography gene={gene as any} style={styles.sectionSeparator} />
        <Separator style={styles.sectionSeparator} />
      </>
    )
  }

  return (
    <StickyTabPageScrollView contentContainerStyle={{ paddingTop: 15 }}>
      {biography()}
      {relatedArtists()}
    </StickyTabPageScrollView>
  )
}

interface Styles {
  sectionSeparator: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  sectionSeparator: {
    marginBottom: 20,
  },
})

export default createFragmentContainer(About, {
  gene: graphql`
    fragment About_gene on Gene {
      ...Biography_gene
      trending_artists: trendingArtists {
        ...RelatedArtists_artists
      }
    }
  `,
})
