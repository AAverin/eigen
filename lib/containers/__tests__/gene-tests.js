import 'react-native'
import React from 'react'
import Relay from 'react-relay'
import renderer from 'react-test-renderer'

import Gene from '../gene'

it('looks like expected', () => {
  const props = {
      gene: {
        id: 'An ID',
        _id: 'a UUID',
        name: 'Example Gene'
      }
    }

  const tree = renderer.create(
    <Gene geneID={props.gene.name} gene={props.gene}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
