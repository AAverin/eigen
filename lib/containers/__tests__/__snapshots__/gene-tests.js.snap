exports[`test looks like expected 1`] = `
<ScrollView
  scrollsToTop={true}>
  <View
    style={
      Object {
        "paddingLeft": 20,
        "paddingRight": 20
      }
    }>
    <View
      style={
        Object {
          "paddingTop": 20
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        lineBreakMode="tail"
        style={
          Array [
            Object {
              "fontSize": 12
            },
            Array [
              Object {
                "textAlign": "center"
              },
              Object {
                "fontSize": 14
              }
            ],
            Object {
              "fontFamily": "Avant Garde Gothic ITCW01Dm"
            }
          ]
        }>
        EXAMPLE GENE
      </Text>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "borderColor": "red",
              "borderWidth": 1
            },
            Array [
              Object {
                "height": 46
              },
              Object {
                "marginBottom": 30,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 30
              }
            ]
          ]
        } />
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          lineBreakMode="tail" />
      </View>
    </View>
  </View>
</ScrollView>
`;
